src/plugins/mpi/pmix/pmixp_info.c

```diff
@@ -118,57 +122,73 @@ bool pmixp_info_srv_fence_coll_barrier(void)
 	return _srv_fence_coll_barrier;
 }
 
+/* srun PMIx lib information */
+int pmixp_srun_info_set(const mpi_plugin_client_info_t *job, char ***env)
+{
+	memset(&_pmixp_srun_info, 0, sizeof(_pmixp_srun_info));
+#ifndef NDEBUG
+	_pmixp_srun_info.proc_info.magic = PMIXP_INFO_MAGIC;
+#endif
+	memcpy(&_pmixp_srun_info.proc_info.step_id, &job->step_id,
+	       sizeof(_pmixp_srun_info.proc_info.step_id));
+	pmixp_info_gen_nspace(job->step_id.job_id, job->step_id.step_id,
+			      _pmixp_srun_info.proc_info.nspace);
+	_pmixp_srun_info.proc_info.lib_tmpdir = NULL;
```

Comment:
```suggestion
	_pmixp_srun_info.proc_info.lib_tmpdir = gen_srun_tmpdr(nspace);
```
tmpdir for srun should be different from the stepd to avoid conflict in case they are on the same node.